scripts/helpers.py

```diff
@@ -620,26 +620,22 @@ def get_conv_factors(sector):
             "Gas Coke": 0.007326,
             "Refinery gas": 0.01375,
             "Coal Tar": 0.007778,
+            "Paraffin waxes": 0.01117,
+            "Ethane": 0.01289,
+            "Oil shale": 0.00247,
         }
     return fuels_conv_toTWh
 
 
 def aggregate_fuels(sector):
```

Comment:
Hi @hazemakhalek,

thanks a lot for this great PR.

I noticed that not all the fuels available in get_conv_factors() are present in aggregate_fuels(). My understanding of the code is that what appears in get_conv_factors should be also "mapped" in aggregate_fuels.

I see in fact, the dictionary built with aggregate_fuels in _build_base_industry_totals.py_

```
    # Fetch the fuel categories from the helpers script
    (
        gas_fuels,
        oil_fuels,
        biomass_fuels,
        coal_fuels,
        heat,
        electricity,
    ) = aggregate_fuels("industry")

    # Create fuel dictionary to use for mapping all fuels to the pypsa representative fuels
    fuel_dict = {
        element: var_name
        for var_name, element_list in [
            ("gas", gas_fuels),
            ("oil", oil_fuels),
            ("biomass", biomass_fuels),
            ("heat", heat),
            ("coal", coal_fuels),
            ("electricity", electricity),
        ]
        for element in element_list
    }
```

is then used to populate the "Carrier" column. Namely:



```
df["carrier"] = df["Commodity"].map(fuel_dict)
```

Thanks a lot and please let me know if my understanding of the code is correct

Reply:
Hey @finozzifa , 

Thanks for the quick review! You're interpretation is correct.

I tested it quickly now and the first part of your suggestion is correct and need to be added. As of now, these fuels are calculated but not added to the aggregate value of the representative fuel. (e.g. Aviation gasoline is not added to Oil)

The second part however is intended, the fuels you show here are given in the dataset in energy units already (Terajoules)
These are handled on the line;

`df.loc[index_energy_TJ, "Quantity_TWh"] = df.loc[index_energy_TJ].apply(
        lambda x: x["Quantity"] / 3600, axis=1)`

The fuels conversion factors are only used to convert from mass or volume to energy.

Reply:
Hi @hazemakhalek ,

thanks for your answer and for your explanation :)

In order words, we should include all entries from _get_conv_factors()_ to _aggregate_fuels()_, whereas it is fine if _aggregate_fuels()_ misses some entries available in _get_conv_factors()_.

Did I get it right?

speak soon,

Fabrizio

Reply:
**Answer from PyPSA-Earth-Sec meeting on May 30th 2024**

All fuels should be present in _aggregate_fuels()_, but not all fuels should be in _get_conv_factors()_.

Thanks @hazemakhalek

Reply:
Great discussion!
To add, some factors in the list may be typos or refuses; for simplicity, the naming may be revised to be normalized and maybe avoid duplicated entries that may introduce noise.

What do you think?

Reply:
Hey Davide, I agree with you in general.
The typos in the data could be handled in a more consistent way to correct the trailing spaces, typos of the source.
If you handle this in a specific way for the power model let me know so we can make it coherent. 

If not, I will start an issue which will be good for as an early task for some new comers.